src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -62,6 +62,28 @@ internal VariableState Clone()
             }
         }
 
+        /// <summary>
+        /// Data recorded for a particular analysis run.
+        /// </summary>
+        internal readonly struct Data
+        {
+            /// <summary>
+            /// Number of entries tracked during analysis.
+            /// </summary>
+            internal readonly int TrackedEntries;
+
+            /// <summary>
+            /// True if analysis was required; false if analysis was optional and results dropped.
+            /// </summary>
+            internal readonly bool RequiredAnalysis;
+
+            internal Data(int trackedEntries, bool requiredAnalysis)
+            {
+                TrackedEntries = trackedEntries;
+                RequiredAnalysis = requiredAnalysis;
+            }
+        }
```

Comment:
Embrace C# 9 :smile: 

```suggestion
        internal readonly struct Data
        {
            /// <summary>
            /// Number of entries tracked during analysis.
            /// </summary>
            internal int TrackedEntries { get; init; }

            /// <summary>
            /// True if analysis was required; false if analysis was optional and results dropped.
            /// </summary>
            internal bool RequiredAnalysis { get; init; }
        }
```

Don't feel strongly about this, just wanted to point out it's an option

Reply:
Thanks. I actually prefer `readonly` fields for internal types.

---
In reply to: [544521768](https://github.com/dotnet/roslyn/pull/49953#discussion_r544521768) [](ancestors = 544521768)

Reply:
Sigh. If only LDM had taken my suggestion to let `init` apply to fields too.